projects/aca-content/src/lib/components/knowledge-retrieval/search-ai/search-ai-results/search-ai-results.component.spec.ts

```diff
@@ -0,0 +1,107 @@
+/*!
+ * Copyright © 2005-2024 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Alfresco Example Content Application
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail. Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { TestBed, ComponentFixture } from '@angular/core/testing';
+import { SearchAiResultsComponent } from './search-ai-results.component';
+import { ActivatedRoute, Params } from '@angular/router';
+import { Subject } from 'rxjs';
+import { MatSnackBarModule } from '@angular/material/snack-bar';
+import { UserPreferencesService } from '@alfresco/adf-core';
+import { AppTestingModule } from '../../../../testing/app-testing.module';
+import { MatDialogModule } from '@angular/material/dialog';
+
+describe('SearchAiResultsComponent', () => {
+  let fixture: ComponentFixture<SearchAiResultsComponent>;
+  let component: SearchAiResultsComponent;
+  let spyPerformAiSearch: jasmine.Spy;
+  let mockQueryParams = new Subject<Params>();
+  let knowledgeRetrievalNodes = '';
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [AppTestingModule, SearchAiResultsComponent, MatSnackBarModule, MatDialogModule],
+      providers: [
+        {
+          provide: ActivatedRoute,
+          useValue: {
+            queryParams: mockQueryParams.asObservable()
+          }
+        },
+        { provide: UserPreferencesService, useValue: { get: () => knowledgeRetrievalNodes } }
+      ]
+    });
+
+    fixture = TestBed.createComponent(SearchAiResultsComponent);
+    component = fixture.componentInstance;
+    spyPerformAiSearch = spyOn(component, 'performAiSearch');
```

Comment:
we should not spy on component's function. We can spy on functions which come from outside of component like services' functions. Instead of spying component's function we should test based on what's happening inside that function.